device_backends/uio/src/UioAccess.cc

```diff
@@ -56,10 +56,11 @@ namespace ChimeraTK {
       throw ChimeraTK::logic_error("UIO: Read request exceeds device memory region");
     }
 
-    void* targetAddress = static_cast<uint8_t*>(_deviceUserBase) + address;
-
-    // Is inherently thread-safe
-    std::memcpy(data, targetAddress, sizeInBytes);
+    volatile int32_t* rptr = static_cast<volatile int32_t*>(_deviceUserBase) + address / sizeof(int32_t);
+    while(sizeInBytes >= sizeof(int32_t)) {
```

Comment:
There are up to three bytes uncopied. I guess this is not a valid use case with UIO? Do we have a check for it?

Reply:
Good question, for our UIO this is correct, probably not for the generic case?